tests/dragonfly/connection_test.py

```diff
@@ -1068,13 +1075,38 @@ async def test_timeout(df_server: DflyInstance, async_client: aioredis.Redis):
 
     await asyncio.sleep(2)
 
-    @assert_eventually
-    async def wait_for_conn_drop():
+    await wait_for_conn_drop(async_client)
+    info = await async_client.info("clients")
+    assert int(info["timeout_disconnects"]) >= 1
+
+
+@dfly_args({"send_timeout": 3})
+async def test_send_timeout(df_server, async_client: aioredis.Redis):
+    reader, writer = await asyncio.open_connection("127.0.0.1", df_server.port)
+    writer.write(f"client setname writer_test\n".encode())
+    clients = await async_client.client_list()
+    assert len(clients) == 2
```

Comment:
you can not be sure your writer really connected, so you should wait for `len(clients) == 2`.
the reason for this is that you send a packet on the socket but you do not know when Dragonfly actually reads and establishes a connection

Reply:
you can also ensure established connection by reading the first response with `assert "OK" in (await reader.readline()).decode()`